dark/dna.py

```diff
@@ -215,3 +216,40 @@ def _ambiguousMatch(a, b, matchAmbiguous):
             'gapOffsets': read2GapOffsets,
         },
     }
+
+
+def findKozakConsensus(read):
+        """
+        In a given DNA sequence, search for a Kozak consensus: (gcc)gccRccATGG.
+        Upper case bases are required, lower case bases are the most frequent
+        bases at the given position. The sequence in brackets is of uncertain
+        significance and is not taken into account here.
+
+        @param read: A C{DNARead} instance to be checked for Kozak consensi.
+
+        @return: A generator that yields the read, id, original Read, start
+        offset, stop offset and quality percentage of the Kozak consensus
+        sequence. Included Kozak consensuses have all required bases. The
+        consensus at non-required bases of significance is given in the
+        quality percentage (x out of 5).
+        """
+        readLen = len(read)
```

Comment:
Test readLen > 9

Reply:
done. 

Also changed line underneath to avoid an IndexError, to:
`while offset < readLen + 3:`

Reply:
Added a test for empty string and a short nt sequence.